### PR TITLE
add saber-data

### PR DESCRIPTION
**Pull request by mmiesch**
## Description

Adding saber-data to the develop branch of the public fv3-bundle

## Definition of Done

fv3-bundle builds and saber tests pass

### Issue(s) addressed

Link the issues to be closed with this PR
None

## Dependencies

None

## Impact

None

### CODE REVIEW

**Review by benjaminmenetrier — APPROVED** [review]

**Comment by mer-a-o** [comment]
During the mpas release, I found a bug in bundle CMakeLists.txt. I pushed a commit to fix this bug.

File: CMakeLists.txt [hunk 1]
```diff
@@ -78,33 +78,43 @@ ecbuild_bundle( PROJECT fv3-jedi    GIT "https://github.com/jcsda/fv3-jedi.git"
 find_branch_name(REPO_DIR_NAME ioda)
 # When LOCAL_PATH_JEDI_TESTFILES is set to the directory of IODA test files stored
 # in a local directory, ioda-data repo will not be cloned
-if( NOT DEFINED ENV{LOCAL_PATH_JEDI_TESTFILES} AND NOT DEFINED ${GIT_TAG_FUNC} )
```
Comment (shlyaeva) [inline CMakeLists.txt:81]: I noticed `${GIT_TAG_FUNC}` is replaced by `GIT_TAG_FUNC`, is the behavior unchanged? (I'm not very familiar with cmake)

File: CMakeLists.txt [hunk 2]
```diff
@@ -78,33 +78,43 @@ ecbuild_bundle( PROJECT fv3-jedi    GIT "https://github.com/jcsda/fv3-jedi.git"
...
+if( NOT DEFINED ENV{LOCAL_PATH_JEDI_TESTFILES} AND NOT DEFINED GIT_TAG_FUNC )
   ecbuild_bundle( PROJECT ioda-data GIT "https://github.com/JCSDA-internal/ioda-data.git" BRANCH develop UPDATE )
-endif()
 
-# If IODA's current branch is available in ioda-data repo, that branch will be checked out
-branch_checkout (REPO_DIR_NAME ioda-data
-                 BRANCH ${GIT_BRANCH_FUNC} )
+  # If IODA's current branch is available in ioda-data repo, that branch will be checked out
+  branch_checkout (REPO_DIR_NAME ioda-data
```
Comment (shlyaeva) [inline CMakeLists.txt:85]: I see this is now inside the `if`. Looks good to me, seems like only need to checkout branch if `ioda-data` is used.

**Review by shlyaeva — APPROVED** [review]
I have one question, see below. Changes look good to me, although I'm not very familiar with this. I did not test, only reviewed the code changes.

File: CMakeLists.txt (hunk 1 above)
Comment (mer-a-o) [reply CMakeLists.txt:81]: It's a weird cmake behavior. We need to use `DEFINED` without `${}` to check if the variable is defined or not.

**Comment by mmiesch** [comment]
Thanks @mer-a-o for the corrections.   If you think it's ready to go, please approve and then I'll merge it.

**Review by srherbener — APPROVED** [review]
I think these changes look good. However, I did not test them.

I noticed that similar changes should probably be made in ioda-bundle. Ie:
* remove the `${}` around the GIT_TAG_FUNC in the `AND NOT DEFINED` expression
* place the `branch_checkout ...` command inside the corresponding if statement

Is this correct? If so I can create a PR. Thanks!

**Review by mer-a-o — APPROVED** [review]

**Comment by mer-a-o** [comment]
@srherbener, yes we will need to update the bundles with these 2 changes.

**Pull request merged**